modin/polars/base.py

```diff
@@ -0,0 +1,711 @@
+        """
+        Explode the given columns to long format.
+
+        Args:
+            columns: Columns to explode.
+            more_columns: Additional columns to explode.
+
+        Returns:
+            DataFrame with the columns exploded.
+        """
+        if len(more_columns) > 0:
+            columns = [columns, *more_columns]
+        return self.__constructor__(
+            _query_compiler=self._query_compiler.explode(columns)
+        )
+
+    def extend(self, other: "BasePolarsDataset") -> "BasePolarsDataset":
+        """
+        Extend the DataFrame with another DataFrame.
+
+        Args:
+            other: DataFrame to extend with.
+
+        Returns:
+            Extended DataFrame for convenience. DataFrame is modified in place.
+        """
+        self._query_compiler = self._query_compiler.concat(
+            axis=0, other=other._query_compiler
+        )
+        return self
+
+    def fill_nan(self, value):
+        """
+        Fill NaN values with the given value.
+
+        Args:
+            value: Value to fill NaN values with.
+
+        Returns:
+            DataFrame with NaN values filled.
+        """
+        # TODO: Handle null values differently than nan.
+        return self.__constructor__(_query_compiler=self._query_compiler.fillna(value))
+
+    def fill_null(
+        self,
+        value: Any | None = None,
+        strategy: str | None = None,
+        limit: int | None = None,
+        *,
+        matches_supertype: bool = True,
+    ) -> "BasePolarsDataset":
+        """
+        Fill null values with the given value or strategy.
+
+        Args:
+            value: Value to fill null values with.
+            strategy: Strategy to fill null values with.
+            limit: Maximum number of null values to fill.
+            matches_supertype: Whether the value matches the supertype.
+
+        Returns:
+            DataFrame with null values filled.
+        """
+        if strategy == "forward":
+            strategy = "ffill"
+        elif strategy == "backward":
+            strategy = "bfill"
+        elif strategy in ["min", "max", "mean"]:
+            value = getattr(self, strategy)()._query_compiler
+            strategy = None
+        elif strategy == "zero":
+            strategy = None
+            value = 0
+        elif strategy == "one":
+            strategy = None
+            value = 1
+        else:
+            raise ValueError(f"Unknown strategy: {strategy}")
+        return self.__constructor__(
+            _query_compiler=self._query_compiler.fillna(
+                value=value, method=strategy, limit=limit
+            )
+        )
+
+    def filter(self, *predicates, **constraints: Any) -> "BasePolarsDataset":
+        if len(predicates) > 1:
+            predicates = predicates[0]
+            for p in predicates[1:]:
+                predicates = predicates & p
+        else:
+            predicates = predicates[0]
```

Comment:
```suggestion
        predicates = predicates[0]
        for p in predicates[1:]:
            predicates = predicates & p
```

nit: this should work